lib/ecto/association.ex

```diff
@@ -412,13 +422,15 @@ defmodule Ecto.Association.Has do
   end
 
   @doc false
-  def on_repo_change(%{on_replace: on_replace} = refl, parent_changeset,
+  def on_repo_change(%{on_replace: on_replace} = refl, %{data: parent} = parent_changeset,
                      %{action: :replace} = changeset, opts) do
     changeset = case on_replace do
       :nilify -> %{changeset | action: :update}
       :delete -> %{changeset | action: :delete}
     end
 
+    changeset = Ecto.Association.update_parent_prefix(changeset, parent)
```

Comment:
This is required here since the parent is nullified on line 434 and thus looses its prefix. I was unsure if you would normally add a comment here to reflect that.